railties/lib/rails/generators/rails/plugin/plugin_generator.rb

```diff
@@ -468,6 +478,14 @@ def relative_path
         return unless inside_application?
         app_path.delete_prefix("#{rails_app_path}/")
       end
+
+      def gemfile_path
+        if inside_application?
+          File.join("..", dummy_path, "Gemfile")
+        else
+          "../Gemfile"
+        end
+      end
```

Comment:
Including `../` doesn't feel right at this level or for a method named `gemfile_path`.  Maybe if named it `gemfile_relative_path` and passed an argument, e.g. `gemfile_relative_path("..")`?

Alternatively, replace the call to this method with something like `File.join("..", *app_relative_path, "Gemfile")`.